examples/embedding/embedding.cpp

```diff
@@ -236,9 +237,24 @@ int main(int argc, char ** argv) {
                 LOG("\n");
             }
         } else if (pooling_type == LLAMA_POOLING_TYPE_RANK) {
+            const uint32_t n_cls_out = llama_model_n_cls_out(model);
+            std::vector<std::string> cls_out_labels;
+
+            for (uint32_t i = 0; i < n_cls_out; i++) {
+                const char * label = llama_model_cls_label(model, i);
+                const std::string label_i = label == nullptr || strlen(label) == 0 ? std::to_string(i) : label;
```

Comment:
nits: this can be done without `strlen`, so we don't have to include cstring:

```cpp
const std::string label_i(label == nullptr ? "" : label);
label_i = label_i.empty() ? std::to_string(i) = label_i;
```